### PR TITLE
[pickers] Localize `TimeClock` meridiem text

### PR DESCRIPTION
**Pull request by LukasTy**
Resolves issue noticed in https://github.com/mui/mui-x/issues/10302#issuecomment-1716439968

Before: https://codesandbox.io/s/sleepy-shtern-ljwzmh?file=/Demo.tsx

After: https://codesandbox.io/s/competent-leftpad-gvphzn?file=/Demo.tsx

### CODE REVIEW

File: packages/x-date-pickers/src/TimeClock/Clock.tsx [hunk 1]
```diff
@@ -359,7 +360,9 @@ export function Clock<TDate>(inProps: ClockProps<TDate>) {
             ownerState={ownerState}
             className={classes.amButton}
           >
-            <Typography variant="caption">AM</Typography>
+            <Typography variant="caption" minWidth={19}>
```
Comment (LukasTy) [inline packages/x-date-pickers/src/TimeClock/Clock.tsx:363]: Without an explicit width, the icon button could become too narrow in certain locales (i.e. Arabic).
Do you have a better suggestion than forcing a `minWidth` to combat that? 🤔

Comment (flaviendelangle) [reply packages/x-date-pickers/src/TimeClock/Clock.tsx:363]: An overflow like we were proposing on the digital clock meridiem section?

Comment (LukasTy) [reply packages/x-date-pickers/src/TimeClock/Clock.tsx:363]: The relevant mentioned issue: https://github.com/mui/mui-x/issues/10031

Comment (LukasTy) [reply packages/x-date-pickers/src/TimeClock/Clock.tsx]: Adjusted to overflow text.
Before: https://codesandbox.io/s/zealous-estrela-chpfnl?file=/Demo.tsx

After: https://codesandbox.io/s/competent-leftpad-gvphzn?file=/Demo.tsx

Comment (flaviendelangle) [reply packages/x-date-pickers/src/TimeClock/Clock.tsx]: For me it make a lot of sense

**Review by flaviendelangle — APPROVED** [review]
Looks great!

File: packages/x-date-pickers/src/TimeClock/Clock.tsx (hunk 1 above)
Comment (noraleonte) [reply packages/x-date-pickers/src/TimeClock/Clock.tsx]: I like the idea of the ellipsis a lot! Makes sense and does not impact the design 🤌

**Review by noraleonte — APPROVED** [review]
Very nice improvement! 🤙

**Pull request merged**